### PR TITLE
hotfix/2.9.2

### PR DESCRIPTION
**Pull request by gjedlicska**
- Update to new specklepy (#1173)
- Fixes liveness and readiness checks to prevent CSRF error message (#1169)
- Fixes broken helm template by adding quotation marks around liveness probe command (#1171)

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [x] I have added appropriate tests.
- [x] I have updated or added relevant documentation.

### CODE REVIEW

File: packages/server/modules/activitystream/services/index.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ module.exports = {
           data: info
         }
       }
-      dispatchStreamEvent({
+      await dispatchStreamEvent({
```
Comment (gjedlicska) [inline packages/server/modules/activitystream/services/index.js:43]: Could this be enough to fix #1141 @didimitrie ?

Comment (gjedlicska) [reply packages/server/modules/activitystream/services/index.js:43]: As a matter of fact, it might not be helping, but the cause is elsewhere.

**Comment by gjedlicska** [comment]
fixes #1141

**Comment by gjedlicska** [comment]
@iainsproat @fabis94 
This includes fixes for #1141 #1171 #1169 and adds ratelimits to the objects rest endpoints.

We need to publish a hotfix `2.9.2` tag before merging to main

File: packages/server/modules/core/rest/download.js [hunk 2]
```diff
@@ -85,6 +93,13 @@ module.exports = (app) => {
     cors(),
     contextMiddleware,
     async (req, res) => {
+      const rejected = await rejectsRequestWithRatelimitStatusIfNeeded({
```
[the remark this answers is not in the document]
Comment (gjedlicska) [reply packages/server/modules/core/rest/download.js:96]: good catch. this should have been propagated to all routes....

File: packages/server/modules/core/rest/diffDownload.js [hunk 3]
```diff
@@ -7,13 +7,22 @@ const { contextMiddleware } = require('@/modules/shared')
...
+  rejectsRequestWithRatelimitStatusIfNeeded
+} = require('@/modules/core/services/ratelimits')
 
 const { pipeline, PassThrough } = require('stream')
 
 module.exports = (app) => {
   app.options('/api/getobjects/:streamId', cors())
 
   app.post('/api/getobjects/:streamId', cors(), contextMiddleware, async (req, res) => {
+    const rejected = await rejectsRequestWithRatelimitStatusIfNeeded({
```
Comment (fabis94) [inline packages/server/modules/core/rest/diffDownload.js:20]: Maybe makes more sense as an Express middleware?

File: packages/server/modules/core/services/ratelimits.js [hunk 4]
```diff
@@ -74,22 +86,31 @@ async function shouldRateLimitNext({ action, source }) {
...
 }
 
+// returns true if the action is fine, false if it should be blocked because of exceeding limit
+async function respectsLimits({ action, source }) {
+  const rateLimitKey = `${action} ${source}`
+  const promise = shouldRateLimitNext({ action, source }).then((shouldRateLimit) => {
+    if (shouldRateLimit) rateLimitedCache[rateLimitKey] = true
+    else delete rateLimitedCache[rateLimitKey]
+  })
+  if (rateLimitedCache[rateLimitKey]) {
```
Comment (fabis94) [inline packages/server/modules/core/services/ratelimits.js:96]: how big does this cache get? should we not store it in redis maybe?

File: packages/server/modules/shared/utils/ip.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const getIpFromRequest = (req) => {
```
Comment (fabis94) [inline packages/server/modules/shared/utils/ip.js:1]: I feel like the concerns of "resolving the request IP" and "filtering out local network IPs" should be separated out (for ease of re-use), but it doesn't matter that much at this point

**Review by fabis94 — APPROVED** [review]

**Pull request merged**